system/services/siteTree/SiteTreeService.cfc

```diff
@@ -847,11 +847,12 @@ component {
 			updated = pobj.updateData(
 				  data                    = data
 				, id                      = arguments.id
-				, useVersioning           = !arguments.skipVersioning
 				, versionNumber           = versionNumber
 				, updateManyToManyRecords = true
 				, forceVersionCreation    = arguments.forceVersionCreation ?: ( pageDataHasChanged || pageTypeDataHasChanged )
 				, isDraft                 = ( arguments.isDraft || pageDataHasChanged || pageTypeDataHasChanged ) ? arguments.isDraft : false
+				, useVersioning           = ( pageDataHasChanged || pageTypeDataHasChanged ) ? !arguments.skipVersioning : false
```

Comment:
I think this begs the question, why are we bothering doing _anything_ if  `!( pageDataHasChanged || pageTypeDataHasChanged )`?